symfony/plugins/orangehrmPimPlugin/Api/OptionalFieldAPI.php

```diff
@@ -0,0 +1,159 @@
+    public function getConfigService(): ConfigService
+    {
+        if (!$this->configService instanceof ConfigService) {
+            $this->configService = new ConfigService();
+        }
+        return $this->configService;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getOne(): EndpointResourceResult
+    {
+        $parameters = [
+            self::PARAMETER_DEPRECATED_FIELDS => $this->getConfigService()->showPimDeprecatedFields(),
+            self::PARAMETER_SIN => $this->getConfigService()->showPimSIN(),
+            self::PARAMETER_SSN => $this->getConfigService()->showPimSSN(),
+            self::PARAMETER_TAX_EXEMPTIONS => $this->getConfigService()->showPimTaxExemptions(),
+        ];
+        return new EndpointResourceResult(ArrayModel::class, $parameters);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForGetOne(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+            new ParamRule(
+                CommonParams::PARAMETER_ID
+            ),
+        );
+    }
+
+    /**
+     * @inheritDoc
+     * @throws Exception
+     */
+    public function update(): EndpointResult
+    {
+        $saveConfig = $this->saveConfig();
+        return new EndpointResourceResult(ArrayModel::class, $saveConfig);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForUpdate(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+            new ParamRule(
+                CommonParams::PARAMETER_ID,
+            ),
+            new ParamRule(self::PARAMETER_DEPRECATED_FIELDS,
+                new Rule(Rules::BOOL_VAL),
+            ),
+            new ParamRule(self::PARAMETER_SIN,
+                new Rule(Rules::BOOL_VAL),
+            ),
+            new ParamRule(self::PARAMETER_SSN,
+                new Rule(Rules::BOOL_VAL),
+            ),
+            new ParamRule(self::PARAMETER_TAX_EXEMPTIONS,
+                new Rule(Rules::BOOL_VAL),
+            ),
+        );
+    }
+
+    /**
+     * @throws CoreServiceException
+     */
+    public function saveConfig(): array
+    {
+        $showSIN = $this->getRequestParams()->getBoolean(RequestParams::PARAM_TYPE_BODY, self::PARAMETER_SIN);
+        $showSSN = $this->getRequestParams()->getBoolean(RequestParams::PARAM_TYPE_BODY, self::PARAMETER_SSN);
+        $showTaxExemptions = $this->getRequestParams()->getBoolean(RequestParams::PARAM_TYPE_BODY,
+            self::PARAMETER_TAX_EXEMPTIONS);
+        $showDeprecatedFields = $this->getRequestParams()->getBoolean(RequestParams::PARAM_TYPE_BODY,
+            self::PARAMETER_DEPRECATED_FIELDS);
+        $this->getConfigService()->setShowPimSSN($showSSN);
+        $this->getConfigService()->setShowPimSIN($showSIN);
+        $this->getConfigService()->setShowPimTaxExemptions($showTaxExemptions);
+        $this->getConfigService()->setShowPimDeprecatedFields($showDeprecatedFields);
+        $parameters = [
```

Comment:
prefer to separate, since duplicated in getOne also
```php
        $parameters = [
            self::PARAMETER_DEPRECATED_FIELDS => $this->getConfigService()->showPimDeprecatedFields(),
            self::PARAMETER_SIN => $this->getConfigService()->showPimSIN(),
            self::PARAMETER_SSN => $this->getConfigService()->showPimSSN(),
            self::PARAMETER_TAX_EXEMPTIONS => $this->getConfigService()->showPimTaxExemptions(),
        ];
```

Reply:
used a private function